### ISSUE TITLE
dotnet test reports more tests

### ISSUE THREAD

**Issue by mattfrear** [adamralph/xbehave.net#551]
Hi

I've been using XBehave for a couple of years, and one thing that's always bugged me is that `dotnet test` reports more tests than there really are. e.g. I've just started a new project, and I have only one test in the project:
```csharp
[Scenario]
public void ShouldReturnProducts()
{
    "Given an address".x(() => fixture.GivenAnAddress());
    "When I request products for the address".x(() => fixture.WhenIRequestProducts());
    "Then the response should be OK".x(() => fixture.ThenTheResponseCodeShouldBeSuccessful());
    "And products should be returned".x(() => fixture.ThenProductsShouldHaveBeenReturned());
}
```

In my VS test runner, this shows up as one test. But when I run `dotnet test` I see:
```
dotnet test
Build started, please wait...
Build completed.

Test run for C:\code\Mobile.Provisioning.AcceptanceTests.dll(.NETCoreApp,Version=v2.2)
Microsoft (R) Test Execution Command Line Tool Version 15.9.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...

Total tests: 4. Passed: 4. Failed: 0. Skipped: 0.
Test Run Successful.
Test execution time: 4.7958 Seconds
```

**Expected output:**
Total tests: 1. Passed: 1. Failed: 0. Skipped: 0.

**Actual output:**
Total tests: 4. Passed: 4. Failed: 0. Skipped: 0.

**Comment by adamralph** [comment]
@mattfrear thanks for asking this question.

This is part of the design of xbehave. E.g. from the repo banner:

> An xUnit.net extension for describing _each step in a test_ with natural language.

xbehave is deliberately built this way so that each step can be reported individually, as succeeded, failed, or skipped.

**Comment by mattfrear** [comment]
Thanks @adamralph for the explanation. 

It does seem a bit strange that it does that, especially when it comes to viewing the test run results in the likes of an Azure DevOps build pipeline - ooh, we have 500 tests cool. Erm, not really, we actually have 50. And it seems different from every other test suite that I know about. Perhaps this is something that could be configurable?

**Comment by adamralph** [comment]
> It does seem a bit strange that it does that

As I said, this is the very purpose of xbehave. If this is changed, there's little or no benefit in using xbehave over regular xunit.

> ooh, we have 500 tests cool. Erm, not really, we actually have 50

I"m not sure that the number of tests tells you anything. It's possible to have poorly tested code with a large number of tests and well tested code with a small number of tests.

> it seems different from every other test suite that I know about

That's the idea. 🙂 

> Perhaps this is something that could be configurable?

If you configured it to generate a single test per scenario, then you would only see success/failure/skip for the entire scenario, and not for each step. At that point, the steps are not giving you any benefit. The only function they serve is to make the test more readable, but you can do that equally well with comments:

```c#
// Given...
var foo = new Foo();

// When...
foo.Bar();
```

So in summary, if you take away "a test per step", you're taking away the very thing that makes xbehave what it is, so you may as well just use regular xunit.
